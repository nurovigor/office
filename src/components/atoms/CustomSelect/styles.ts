import { css } from 'linaria';
import { colors } from 'src/components/_settings/_variables';

/*export const Container = css`
	margin-right: 10px;
	cursor: pointer;
`;

export const Select = css`
  border: 1px solid ${colors.dressBlues};
  background: ${colors.whiteSmokeColor};
  padding: 5px 10px;
	color: ${colors.primaryColor}
	font-weight: 700;
	&:focus-visible{
		outline: none;
	}
`;*/

export const Container = css`
	margin-right: 10px;
	cursor: pointer;
	width: 150px;
`;

export const Select = css`
	height: 29px;
	background-color: ${colors.primaryColor};
	color: ${colors.white};
	font-size: 13px;
	font-weight: 700;
	display: flex;
	align-items: center;
	padding: 2px 10px;
	position: relative;
	&:after {
		content: '';
		background-image: url('../../../assets/up-arrow.png');
		background-position: center;
		background-size: contain;
		background-repeat: no-repeat;
		color: ${colors.white};
		width: 30px;
		height: 30px;
		display: flex;
		justify-content: space-between;
		text-align: center;
		transition: all 0.4s ease;
		position: absolute;
		z-index: 100;
		right: 0;
	}
`;

export const List = css`
	position: absolute;
	box-sizing: border-box;
	background-color: ${colors.white};
	z-index: 10;
	top: 21px;
	right: 0;
	list-style: none;
	padding: 0;
	color: ${colors.blackColor};
	width: 150px;
	border: 1px solid ${colors.primaryColor};
`;

export const Active = css`
	&:after {
		transform: rotateZ(-180deg);
	}
`;

export const Opt = css`
	padding: 10px;
	&:hover {
		background-color: ${colors.success};
		color: ${colors.white};
	}
`;
