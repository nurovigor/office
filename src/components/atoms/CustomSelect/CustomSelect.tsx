import React, { useRef, useState } from 'react';
import { cx } from '@linaria/core';
import { Active, Container, List, Opt, Select } from './styles';

type CustomSelectV2Props = {
	optionsList: string[];
	onChangeOption: (filterName: string, option: string) => void;
	filterName: string;
	selectedFilter: string;
};

export const CustomSelect: React.FC<CustomSelectV2Props> = ({
	optionsList,
	selectedFilter,
	filterName,
	onChangeOption
}) => {
	const [isOptionsOpen, setIsOptionsOpen] = useState<boolean>(false);

	const ref = useRef<HTMLTableRowElement>(null);

	/*useEffect(() => {
		const checkIfClickedOutside = (e: MouseEvent) => {
			// @ts-ignore
			if (isOptionsOpen && ref.current && !ref.current.contains(e.target)) {
				setIsOptionsOpen(false);
			}
		};

		document.addEventListener('mousedown', checkIfClickedOutside);

		return () => {
			document.removeEventListener('mousedown', checkIfClickedOutside);
		};
	}, [isOptionsOpen]);*/

	const onClickHandle = () => {
		setIsOptionsOpen(!isOptionsOpen);
	};

	return (
		<div className={Container} ref={ref}>
			<div
				data-testid="show-options"
				onClick={onClickHandle}
				className={cx(Select, isOptionsOpen && Active)}
			>
				{selectedFilter ? selectedFilter : `Filter by ${filterName}`}
				{isOptionsOpen && (
					<ul className={List}>
						<li
							data-testid="select-option"
							className={Opt}
							onClick={() => onChangeOption(filterName, '')}
						>
							Filter by {filterName}
						</li>
						{optionsList.map((option, index) => (
							<li className={Opt} key={index} onClick={() => onChangeOption(filterName, option)}>
								{option}
							</li>
						))}
					</ul>
				)}
			</div>
		</div>
	);
};
