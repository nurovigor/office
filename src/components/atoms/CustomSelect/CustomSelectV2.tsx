import React, { useEffect, useRef, useState } from 'react';
import { Active, Container, List, Opt, Select } from './styles';
import { cx } from '@linaria/core';

type CustomSelectV2Props = {
	optionsList: string[];
	onChangeOption: (filterName: string, option: string) => void;
	filterName: string;
	selectedFilter: string;
};

export const CustomSelectV2: React.FC<CustomSelectV2Props> = ({
	optionsList,
	selectedFilter,
	filterName,
	onChangeOption
}) => {
	const [isOptionsOpen, setIsOptionsOpen] = useState<boolean>(false);

	const ref = useRef<HTMLTableRowElement>(null);

	useEffect(() => {
		const checkIfClickedOutside = (e: MouseEvent) => {
			// @ts-ignore
			if (isOptionsOpen && ref.current && !ref.current.contains(e.target)) {
				setIsOptionsOpen(false);
			}
		};

		document.addEventListener('mousedown', checkIfClickedOutside);

		return () => {
			document.removeEventListener('mousedown', checkIfClickedOutside);
		};
	}, [isOptionsOpen]);

	const onClickHandle = () => {
		setIsOptionsOpen(!isOptionsOpen);
	};

	return (
		<div className={Container} ref={ref}>
			<div onClick={onClickHandle} className={cx(Select, isOptionsOpen && Active)}>
				{selectedFilter ? selectedFilter : `Filter by ${filterName}`}
				{isOptionsOpen && (
					<ul className={List}>
						<li className={Opt} onClick={() => onChangeOption(filterName, '')}>
							Filter by {filterName}
						</li>
						{optionsList.map((option, index) => (
							<li className={Opt} key={index} onClick={() => onChangeOption(filterName, option)}>
								{option}
							</li>
						))}
					</ul>
				)}
			</div>
		</div>
	);
};
