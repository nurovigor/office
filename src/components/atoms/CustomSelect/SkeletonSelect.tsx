import React from 'react';
import { SkeletonDiv } from 'src/components/atoms/SkeletonDiv';
import { Preloader } from 'src/components/atoms/CustomSelect/styles';

export const SkeletonSelect = () => {
	return (
		<div className={Preloader}>
			<SkeletonDiv width={150} height={33} />
			<SkeletonDiv width={150} height={33} />
			<SkeletonDiv width={150} height={33} />
		</div>
	);
};
