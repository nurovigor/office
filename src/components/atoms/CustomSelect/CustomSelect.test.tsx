import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { CustomSelect } from 'src/components/atoms/CustomSelect/CustomSelect';

describe('CustomSelect component', () => {
	let options: string[];
	let onChange: (filterName: string, option: string) => void;
	let filterName: string;
	let selectedFilter: string;
	beforeEach(() => {
		options = ['name', 'type', 'bind'];
		onChange = jest.fn();
		filterName = 'name';
		selectedFilter = 'Panasonic';
	});
	it('renders CustomSelect', () => {
		const { asFragment } = render(
			<CustomSelect
				optionsList={options}
				onChangeOption={onChange}
				filterName={filterName}
				selectedFilter={selectedFilter}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('filter by Panasonic', () => {
		render(
			<CustomSelect
				optionsList={options}
				onChangeOption={onChange}
				filterName={filterName}
				selectedFilter={selectedFilter}
			/>
		);
		expect(screen.getByText('Panasonic')).toBeInTheDocument();
	});

	it('filter not selected', () => {
		const { asFragment } = render(
			<CustomSelect
				optionsList={options}
				onChangeOption={onChange}
				filterName={filterName}
				selectedFilter={''}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('onChange called with params', () => {
		render(
			<CustomSelect
				optionsList={options}
				onChangeOption={onChange}
				filterName={filterName}
				selectedFilter={selectedFilter}
			/>
		);
		fireEvent.click(screen.getByTestId('show-options'));
		fireEvent.click(screen.getByTestId('select-option'));
		expect(onChange).toBeCalledWith('name', '');
	});

	it('renders CustomSelect with open options', () => {
		const { asFragment } = render(
			<CustomSelect
				optionsList={options}
				onChangeOption={onChange}
				filterName={filterName}
				selectedFilter={selectedFilter}
			/>
		);
		fireEvent.click(screen.getByTestId('show-options'));
		expect(asFragment()).toMatchSnapshot();
	});
});
