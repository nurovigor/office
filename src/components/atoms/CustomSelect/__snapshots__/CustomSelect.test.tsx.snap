// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect component filter not selected 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="show-options"
    >
      Filter by name
    </div>
  </div>
</DocumentFragment>
`;

exports[`CustomSelect component renders CustomSelect 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="show-options"
    >
      Panasonic
    </div>
  </div>
</DocumentFragment>
`;

exports[`CustomSelect component renders CustomSelect with open options 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="show-options"
    >
      Panasonic
      <ul>
        <li
          data-testid="select-option"
        >
          Filter by name
        </li>
        <li>
          name
        </li>
        <li>
          type
        </li>
        <li>
          bind
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
